src/cuda/cuda_transformer.h

```diff
@@ -147,22 +147,19 @@ class CudaTransformer : public CfgVisitor
         return add_transfer_function(x, vars);
     }
 
-    void fill_with_variable_indices(int* arr, std::set<std::string>& vars){
-        int i = 0;
+    BitVector fill_with_variable_indices(std::set<std::string>& vars){
+        BitVector result = 0;
         for(std::string var : vars){
             if (variables.count(var)){
-                arr[i++] = variables[var];
+                result |= 1<<variables[var];
             }
-        } 
+        }
+        return result;
     }
```

Comment:
function signature has been changed so maybe `get_variable_indices` is a better name now?